ulib/FStar.Matrix.fst

```diff
@@ -410,12 +521,30 @@ let matrix_mul_gen #c #eq #m #n #p (add mul: CE.cm c eq)
 
 let matrix_mul #c #eq #m #n #p (add mul: CE.cm c eq) (mx: matrix c m n) (my: matrix c n p) 
   = init (matrix_mul_gen add mul mx my)
- 
-let seq_last_index #c (s: SB.seq c{SB.length s > 0}) : Lemma (SProp.last s == SB.index s (SB.length s - 1)) = ()
 
+(* the following lemmas improve verification performance. *) 
+(* Sometimes this fact gets lost and needs an explicit proof *)
+let seq_last_index #c (s: SB.seq c{SB.length s > 0}) 
+  : Lemma (SProp.last s == SB.index s (SB.length s - 1)) = ()
+
+(* It often takes assert_norm to obtain the fact that,
+   (fold s == last s `op` fold (slice s 0 (length s - 1))).
+   Invoking this lemma instead offers a more stable option. *)
 let seq_fold_decomposition #c #eq (cm: CE.cm c eq) (s: SB.seq c{SB.length s > 0}) 
   : Lemma (SP.foldm_snoc cm s == cm.mult (SProp.last s) (SP.foldm_snoc cm (fst (SProp.un_snoc s)))) = ()
-  
+
+
+(* 
+   A little (maybe silly) word of warning to all who might consider
```

Comment:
Can you please remove this part about redefining `=` warnings ... since it is not related to the module itself, we can keep it out of here. You can add it to some wiki page (I believe there's one on operator overloading). You may keep the comment at line 544-545 if you wish.

Reply:
ok will do